libass/ass_render.c

```diff
@@ -67,6 +68,22 @@ static void text_info_done(TextInfo* text_info)
     free(text_info->combined_bitmaps);
 }
 
+#if CONFIG_PTHREAD
+unsigned default_threads(void)
+{
+#ifdef _SC_NPROCESSORS_ONLN
+    long sc = sysconf(_SC_NPROCESSORS_ONLN);
+    if (sc < 0)
+        return 1;
+    return sc;
```

Comment:
The existing code matches what libc++, ffmpeg, and plenty of other projects do, so I think it's fine for a default.

Reply:
iiuc `sched_getaffinity` may give us better results here and now ffmpeg prefers this too. However, it’s Linux-specific so we’d still need to fallback to `_ONLN` or `_CONF` for other systems (or too old Linux/libc). Notably Android apparently doesn’t or didn’t use to expose a `sched_getaffinity` function (but supports the syscall) but i don’t know how much we should workaround limited platform API.

Reply:
Looks like `_SC_NPROCESSORS_ONLN` is sometimes implemented in terms of `SYS_sched_getaffinity` (musl does this all the time, and glibc uses it as a fallback), and the main difference in practice is that `sched_getaffinity` will exclude cores that the caller has explicitly excluded the process from using (via `sched_setaffinity`)? Reasonable enough, I suppose; should be simple enough to check for this function and use it if available.

Note that Android _does_ expose `sched_getaffinity`, and apparently has since 12 (which is _extremely_ ancient).

Reply:
> Note that Android does expose sched_getaffinity

neat

> since 12 (which is extremely ancient).

Android 12 is only a year older than this PR ^^`
Albeit support for 11 ended and ideally nobody would (need to) use systems without security support, the [first search result](https://gs.statcounter.com/os-version-market-share/android) claims versions <12 are still make up over 30% of the install base

Reply:
Sorry, I was unclear, I meant _API level 12_, which is Android 3. [_Very_ ancient.](https://apilevels.com)

Reply:
oh, yeah Android 3 really is ancient

*(Wondering why some excluse `sched_getaffinity` on Android or use raw syscalls for it, it found this [old bug](https://web.archive.org/web/20130701075313/http://code.google.com/p/android/issues/detail?id=19851#c2) but i sure hope this already got fixed over the last 12 years)*